google-cloud-spanner/src/test/java/com/google/cloud/spanner/DatabaseClientImplTest.java

```diff
@@ -909,7 +909,7 @@ public void testTransactionManager() {
           assertNotNull(manager.getCommitTimestamp());
           break;
         } catch (AbortedException e) {
-          transaction = manager.resetForRetry();
```

Comment:
These assignments are not unnecessary. They re-assign a new transaction context after an abort. Without this assignment, the transaction context that was aborted will be re-used during the retry (note that it is in a `while(true)` loop).

Reply:
I think I might be missing something here, because the first line of the `while` loop is a `manager.begin()`, which will re-assign the `transaction` variable, so the `transaction` assigned in the `manager.resetForRetry()` would be thrown away?

Reply:
Oeps. You're right about that. The problem in this case is actually that the `transaction.begin()` statement is inside the while-loop. That call is supposed to be outside the while-loop. I'll open a separate issue for that and check it in other places as well. That means that this will probably change soon again, but let's just keep this as it is for now, to prevent the code analyzer from complaining about it.

Reply:
Cool, thanks!